pippo-server-parent/pippo-undertow/src/main/java/ro/pippo/undertow/websocket/UndertowWebSocketAdapter.java

```diff
@@ -48,6 +49,7 @@
     private static final Logger log = LoggerFactory.getLogger(UndertowWebSocketAdapter.class);
 
     private static List<WebSocketConnection> connections = new CopyOnWriteArrayList<>();
+    private static List<WebSocketConnection> connectionsReadOnly = Collections.unmodifiableList(connections);
```

Comment:
It's for the first time when I see this construction in the context of dynamic `connections` variable. I was wondering if I modified the `connections` list if the modification is propagated to `connectionsReadOnly`. And the response is yes. So, the code works perfect.
For more info see [this](https://stackoverflow.com/questions/15111073/collections-unmodifiablelist-and-defensive-copy) discussion.